week-2/oo_design.md

```diff
@@ -1,62 +1,97 @@
+---
+
+## Object-Oriented Principles
+
+Object-oriented languages are built on core principles.
+
+- Encapsulation of an Object's **data** and **behavior**, by exposing an interface.
+- Abstraction of complexity, by hiding **implementation details**.
+- Inheritance of common behavior, from a **parent Class** that **shares methods**.
+- Polymorphism of behavior, by responding to **messages by name** and **argument type**.
 
 ---
 
-# Principles of OO Design
+## Encapsulation
 
-Object oriented languages are built on several core principles which we can apply to our JavaScript code
+Encapsulation is the act of keeping data within an object as properties, and using methods to access that data.
 
-- encapsulation of data and behavior
-- pointers
-- privacy of data
+There are some guidelines to keep in mind when using Objects that follow object-oriented design.
 
 ---
 
-# Encapsulation
+## Encapsulation Example
+
+By using the `.describe()` method of the `dog`, you are accessing the data within the dog by using its **public API**. This is opposed to going **inside** the dog and getting the data for yourself.
+
+```js
+const fido = {
+  name: 'fido',
+  color: 'brown',
+  describe () { 
+    return `Hello! My name is ${this.name} and I am ${this.color}.`
+  }
+}
+
+// RECOMMENDED
+console.log(fido.describe());
+
+// DISCOURAGED
+console.log(`Hello! My name is ${fido.name} and I am ${fido.color}`);
+```
+
+---
+
+## Encapsulation Question
+
+> Why would you want to use the public API of the dog object, instead of reaching inside the object the the data you may need?
+
+Example code from prior slide:
 
-"Encapsulation of data and behavior" is really just a fancy way of saying "object"
+```js
+console.log(fido.describe());
 
-BUT there are some guidelines we want to keep in mind when creating objects if we're trying to follow OO design principles
+console.log(`Hello! My name is ${this.name} and I am ${this.color}`);
+```
 
 ---
 
-# The Single Responsibility Principle
+## Single Responsibility Principle
 
 Each object should have limited, clearly defined responsibilities.
 
 > Do one thing, and do it well.
 
 This makes it easier to know *where* in the code to look when you want to add a feature or fix a bug.
 
-It helps to *increase cohesion* and *reduce coupling*... If there are two objects with related (but separate) responsibilities, you can change the implementation of one without affecting the other. 
+It helps to *increase cohesion* and *reduce coupling*... If there are two objects with related (but separate) responsibilities, you can change the implementation of one without affecting the other.
 
 Objects can communicate through the same stable *interface* without regard for the *implementation*.
 
 ---
 
-# Shallow Hierarchies
+## Shallow Hierarchies
 
-OO systems rely on pointers, so there's a risk of confusing designs, including:
+Object-Oriented systems can , so there's a risk of confusing designs, including:
```

Comment:
@Dangeranger I'm not sure how to finish this initial clause here. Did you mean to say they can point? Thanks.